tests/integration/components/form-controls/component-test.ts

```diff
@@ -0,0 +1,83 @@
+import { render } from '@ember/test-helpers';
+import a11yAudit from 'ember-a11y-testing/test-support/audit';
+import Changeset from 'ember-changeset';
+import lookupValidator from 'ember-changeset-validations';
+import setupMirage from 'ember-cli-mirage/test-support/setup-mirage';
+import { setupRenderingTest } from 'ember-qunit';
+import hbs from 'htmlbars-inline-precompile';
+import { module, test } from 'qunit';
+
+import { click } from 'ember-osf-web/tests/helpers';
+
+import { nodeValidation } from './validation';
+
+module('Integration | Component | form-controls', hooks => {
+    setupRenderingTest(hooks);
+    setupMirage(hooks);
+
+    test('it renders 991', async function(assert) {
+        const model = {
+            title: '',
+            description: '',
+        };
+        const changeset = new Changeset(model, lookupValidator(nodeValidation), nodeValidation);
+        this.set('changeset', changeset);
+
+        await render(hbs`
+            <form data-test-form>
+                <FormControls as |form| >
+                    <form.text @label='title' @valuePath='title' />
+                    <form.text @label='description' @valuePath='description' />
+                </FormControls>
+            </form>
+        `);
+        assert.dom('[data-test-form]').exists();
+        assert.dom('.form-group > input').exists({ count: 2 });
+        await a11yAudit(this.element);
+        assert.ok(true, 'No a11y errors on page');
+    });
+
+    test('it validates 991', async function(assert) {
+        const model = {
+            title: '',
+            description: '',
+        };
+        const changeset = new Changeset(model, lookupValidator(nodeValidation), nodeValidation);
+        this.set('changeset', changeset);
+
+        function submit() {
+            // @ts-ignore 'changeset-validations aren't set up correctly for this to typecheck'
```

Comment:
You can drop this ts-ignore and get proper type safety by casting `Changeset` to `ChangesetDef` as I suggested above. The auto-generated types for `ember-changeset` are actually correct, it's just that `ember-changeset` uses a factory pattern that TypeScript can't support (yet). Here are some relevant issues:
https://github.com/Microsoft/TypeScript/issues/11588
https://github.com/Microsoft/TypeScript/issues/27594

Reply:
Would it be worth a util function to avoid manual casting every time we build a changeset?

```ts
function buildChangeset(...args: ConstructorParameters<typeof Changeset>) {
  return new Changeset(...args) as ChangesetDef;
}
```

Reply:
Seems reasonable. Also, if we're going to consistently use `ember-changeset-validations` (which I think we are), we could have this function handle `lookupValidator()`:
```ts
import lookupValidator from 'ember-changeset-validations';
import { Config, ValidatorFunc } from 'ember-changeset/types';

type ValidationMap = { [s: string]: ValidatorFunc | ValidatorFunc[] };

function buildChangeset(target: object, validationMap: ValidationMap, config?: Config) {
  return new Changeset(target, lookupValidator(validationMap), validationMap) as ChangesetDef;
}
```
so, in this case would be used as:
```ts
const changeset = buildChangeset(model, nodeValidation);
```

Reply:
This sounds like a good idea to me.  I'll add it to the changes :+1: